### PR TITLE
Review fixes and refactor:

### PR DESCRIPTION
**Pull request by mtrippled**
1) refactor filtering of validations to specifically avoid
   concurrent checkAccept() calls for the same validation hash.
2) Log when duplicate concurrent inbound ledger and validation requests
   are filtered.
3) RAII for containers that track concurrent inbound ledger and
   validation requests.
4) Comment on when to asynchronously acquire inbound ledgers, which
   is possible to be always OK, but should have further review.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Performance (increase or change in throughput and/or latency)
- [ ] Tests (you added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation update
- [ ] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)
- [ ] Release

### API Impact

- [ ] Public API: New feature (new methods and/or new fields)
- [ ] Public API: Breaking change (in general, breaking changes should only impact the next api_version)
- [ ] `libxrpl` change (any change that may affect `libxrpl` or dependents of `libxrpl`)
- [ ] Peer protocol change (must be backward compatible or bump the peer protocol version)

- - -
This change is [Reviewable](https://reviewable.io/reviews/ximinez/rippled/6)

### CODE REVIEW

File: src/ripple/app/consensus/RCLValidations.h [hunk 1]
```diff
@@ -25,12 +25,16 @@
...
 #include <ripple/protocol/STValidation.h>
+#include <optional>
+#include <set>
 #include <vector>
 
 namespace ripple {
 
 class Application;
 
+enum class BypassAccept { FALSE, TRUE };
```
Comment (ximinez) [inline src/ripple/app/consensus/RCLValidations.h]: The `FALSE` and `TRUE` keywords are `#define`d on Windows, which messes up building this.
```suggestion
enum class BypassAccept : bool { no = false, yes };
```

**Review by bachase — APPROVED** [review]
LGTM as a hotfix

File: src/ripple/app/ledger/impl/InboundLedgers.cpp [hunk 2]
```diff
@@ -156,11 +157,26 @@ class InboundLedgersImp : public InboundLedgers
...
+                << ": " << e.what();
+        }
+        catch (...)
+        {
+            JLOG(j_.warn())
+                << "Unknown exception thrown for acquiring new inbound ledger "
+                << hash;
+        }
         lock.lock();
         pendingAcquires_.erase(hash);
```
Comment (ximinez) [inline src/ripple/app/ledger/impl/InboundLedgers.cpp:181]: The manual calls to `unlock` and `lock` still being here are rubbing me the wrong way. What about:
```suggestion
        try
        {
            {
                std::unique_lock lock(acquiresMutex_);
                if (pendingAcquires_.contains(hash))
                    return;
                pendingAcquires_.insert(hash);
            }
            acquire(hash, seq, reason);
        }
        catch (std::exception const& e)
        {
            JLOG(j_.warn())
                << "Exception thrown for acquiring new inbound ledger " << hash
                << ": " << e.what();
        }
        catch (...)
        {
            JLOG(j_.warn())
                << "Unknown exception thrown for acquiring new inbound ledger "
                << hash;
        }
        std::unique_lock lock(acquiresMutex_);
        pendingAcquires_.erase(hash);
```

**Review by ximinez — CHANGES_REQUESTED** [review]

File: src/ripple/app/consensus/RCLValidations.h (hunk 1 above)
Comment (ximinez) [reply src/ripple/app/consensus/RCLValidations.h]: I pushed this change to the branch so I can go ahead and merge this pull request. and let testing proceed.

**Review by ximinez — APPROVED** [review]

**Review by ximinez — APPROVED** [review]
Pending CI completion

File: src/ripple/app/ledger/impl/InboundLedgers.cpp (hunk 2 above)
Comment (mtrippled) [reply src/ripple/app/ledger/impl/InboundLedgers.cpp:181]: Sure, go ahead. Can also be lock_guard in that case if you want.

Comment (ximinez) [reply src/ripple/app/ledger/impl/InboundLedgers.cpp:181]: I'm fine with leaving this as-is for now, but check out https://github.com/ximinez/rippled/commit/7a247645899edefdccf93eae404a79aacd68eb95

**Pull request merged**